frontend/packages/pipelines-plugin/src/components/pipelines/detail-page-tabs/utils.ts

```diff
@@ -1,9 +1,31 @@
 import * as _ from 'lodash';
-import { TektonParam } from '../../../types';
+import { getResourceModelFromTaskKind } from '../../../utils/pipeline-augment';
+import { PipelineKind, PipelineTask, TektonParam } from '../../../types';
+import { ResourceModelLink } from '../resource-overview/DynamicResourceLinkList';
 
 export const removeEmptyDefaultFromPipelineParams = (parameters: TektonParam[]): TektonParam[] =>
   _.map(
     parameters,
     (parameter) =>
       _.omit(parameter, _.isEmpty(parameter.default) ? ['default'] : []) as TektonParam,
   );
+
+export type PipelineTaskLinks = {
+  taskLinks: ResourceModelLink[];
+  finallyTaskLinks?: ResourceModelLink[];
+};
+
+export const getPipelineTaskLinks = (pipeline: PipelineKind): PipelineTaskLinks => {
+  const toResourceLinkData = (tasks: PipelineTask[]): ResourceModelLink[] =>
+    tasks
+      ?.filter((pipelineTask) => !!pipelineTask.taskRef)
+      ?.map((task) => ({
+        model: getResourceModelFromTaskKind(task.taskRef.kind),
+        name: task.taskRef.name,
+        displayName: task.name,
+      }));
+  return {
+    taskLinks: toResourceLinkData(pipeline.spec.tasks),
+    finallyTaskLinks: toResourceLinkData(pipeline.spec.finally),
```

Comment:
pipeline.spec.finally is optional, if a pipeline doesn't have finally task it will return undefined, but your return type of the function is says finallyTaskLinks will return ResourceModelLink[]. So this should return [] array IMO